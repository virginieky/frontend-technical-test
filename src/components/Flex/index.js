import styled from 'styled-components';
import PropTypes from 'prop-types';

const Flex = styled.div`
  display: flex;
  align-items: ${({ alignItems }) => alignItems};
  flex-direction: ${({ flexDirection }) => flexDirection};
  flex-wrap: ${({ flexWrap }) => flexWrap};
  justify-content: ${({ justifyContent }) => justifyContent};
`;

Flex.defaultProps = {
  alignItems: 'normal',
  flexDirection: 'row',
  flexWrap: 'nowrap',
  justifyContent: 'normal',
};

Flex.propTypes = {
  alignItems: PropTypes.string,
  flexDirection: PropTypes.string,
  flexWrap: PropTypes.string,
  justifyContent: PropTypes.string,
};

export default Flex;
