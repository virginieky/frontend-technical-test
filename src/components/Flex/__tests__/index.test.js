import React from 'react';
import { render } from '@testing-library/react';
import Flex from '../index';

describe('components | Flex', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Flex />);

    expect(asFragment()).toMatchSnapshot();
  });
});
