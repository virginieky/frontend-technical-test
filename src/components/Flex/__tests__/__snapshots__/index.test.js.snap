// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | Flex should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Flex-sc-1vrf9e3-0 kdGXrE"
  />
</DocumentFragment>
`;
